src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -202,8 +203,8 @@ export class MockHttpSocket extends MockSocket {
     }
 
     // If the developer destroys the socket, destroy the original connection.
-    this.once('error', (error) => {
-      socket.destroy(error)
+    this.once('error', () => {
+      socket.destroy()
     })
```

Comment:
Forwarding the error to the `socket` leads to the `error` event handler being called twice.